### PR TITLE
Apply random policy to driver/monitor

### PR DESCRIPTION
**Pull request by yodalee**

### CODE REVIEW

**Comment by yodalee** [comment]
update to latest.

File: common/include/callback.h [hunk 1]
```diff
@@ -74,10 +74,16 @@ template <typename SC_TYPE> class Monitor : public Callback {
   }
 
   void before_clk() {
-    // Always received monitor
-    if (valid == 1 && ready == 1) {
-      const SC_TYPE out = read_func();
+    if (ready == 0) {
+      if (GetRandom()) {
+        ready = 1;
```
Comment (johnjohnlin) [inline common/include/callback.h:79]: ready = GetRandom() is shorter?

File: common/include/callback.h [hunk 2]
```diff
@@ -74,10 +74,16 @@ template <typename SC_TYPE> class Monitor : public Callback {
...
+      }
+    } else if (valid == 1 && ready == 1) {
```
Comment (johnjohnlin) [inline common/include/callback.h:81]: else if (valid == 1) is shorter

File: common/include/callback.h (hunk 1 above)
Comment (yodalee) [reply common/include/callback.h]: fixed.

File: common/include/callback.h (hunk 2 above)
Comment (yodalee) [reply common/include/callback.h]: fixed.

**Pull request merged**